### PR TITLE
pkg/pbutil: add marshal-related tests

### PR DESCRIPTION
**Pull request by yichengq**
for #1467

### CODE REVIEW

File: pkg/pbutil/pbutil_test.go [hunk 1]
```diff
@@ -38,3 +58,20 @@ func TestGetBool(t *testing.T) {
 		}
 	}
 }
+
+type resMarshaler struct {
```
Comment (xiang90) [inline pkg/pbutil/pbutil_test.go]: type dataMarshaler

File: pkg/pbutil/pbutil_test.go [hunk 2]
```diff
@@ -38,3 +58,20 @@ func TestGetBool(t *testing.T) {
...
+	data []byte
+}
+
+func (m *resMarshaler) Marshal() ([]byte, error) {
+	return m.data, nil
+}
+
+type resUnmarshaler struct {
```
Comment (xiang90) [inline pkg/pbutil/pbutil_test.go]: dataUnmarshler

File: pkg/pbutil/pbutil_test.go [hunk 3]
```diff
@@ -16,7 +16,27 @@
...
-import "testing"
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMarshaler(t *testing.T) {
+	data := []byte("test data")
+	m := &resMarshaler{data: data}
+	if g := MustMarshal(m); !reflect.DeepEqual(g, data) {
```
Comment (xiang90) [inline pkg/pbutil/pbutil_test.go]: test panic

File: pkg/pbutil/pbutil_test.go [hunk 4]
```diff
@@ -16,7 +16,27 @@
...
+		t.Errorf("data = %s, want %s", g, m)
+	}
+}
+
+func TestUnmarshaler(t *testing.T) {
+	data := []byte("test data")
+	m := &resUnmarshaler{}
+	MustUnmarshal(m, data)
+	if !reflect.DeepEqual(m.data, data) {
```
Comment (xiang90) [inline pkg/pbutil/pbutil_test.go]: test panic

**Comment by yichengq** [comment]
all comments resolved.

File: pkg/pbutil/pbutil_test.go [hunk 5]
```diff
@@ -38,3 +79,36 @@ func TestGetBool(t *testing.T) {
...
+
+type dataMarshaler struct {
+	data []byte
+}
+
+func (m *dataMarshaler) Marshal() ([]byte, error) {
+	return m.data, nil
+}
+
+type errMarshaler struct {
```
Comment (xiang90) [inline pkg/pbutil/pbutil_test.go]: combine data marshaler and err marshaler

File: pkg/pbutil/pbutil_test.go [hunk 6]
```diff
@@ -38,3 +79,36 @@ func TestGetBool(t *testing.T) {
...
+	err error
+}
+
+func (m *errMarshaler) Marshal() ([]byte, error) {
+	return nil, m.err
+}
+
+type dataUnmarshaler struct {
```
Comment (xiang90) [inline pkg/pbutil/pbutil_test.go]: combine dataunmarshaler and errUnmarshaler

File: pkg/pbutil/pbutil_test.go (hunk 5 above)
Comment (yichengq) [inline pkg/pbutil/pbutil_test.go]: i don't know why i split them.. will do

**Comment by yichengq** [comment]
@xiang90 refreshed

**Comment by xiang90** [comment]
lgtm

**Pull request merged**